x/ccv/provider/types/genesis.go

```diff
@@ -49,12 +49,26 @@ func (gs GenesisState) Validate() error {
 	return nil
 }
 
```

Comment:
For example, `valset_update_id` must be `> 0`.

Reply:
I don't think we need to treat the two cases different. We just need to make sure that all the data in there is valid and it will not lead to the provider crashing.

Reply:
I'd say that without cases distinction we can't validate all the fields depending on their zero value. E.g. `valset_update_id` as an `uint64` can be valid and invalid when set to `0`.

Reply:
It should always be larger than `0`. Why do we need it to be `0`?

Reply:
Bc during a provider chain fresh start,  the CCV `GenesisState` looks like this:
```
&types.GenesisState{ValsetUpdateId:0x0, ConsumerStates:[]types.ConsumerState{}, UnbondingOps:[]types.UnbondingOp{}, MatureUnbondingOps:(*types.MaturedUnbondingOps)(nil), ValsetUpdateIdToHeight:[]types.ValsetUpdateIdToHeight{}, ConsumerAdditionProposals:[]types.ConsumerAdditionProposal{}, ConsumerRemovalProposals:[]types.ConsumerRemovalProposal{}, Params:types.Params{TemplateClient:(*types.ClientState)(0x140013b7320), TrustingPeriodFraction:2, CcvTimeoutPeriod:2419200000000000, InitTimeoutPeriod:604800000000000, VscTimeoutPeriod:3024000000000000, SlashMeterReplenishPeriod:3600000000000, SlashMeterReplenishFraction:"0.05", MaxPendingSlashPackets:1000}}
```

Reply:
That's not good. It should always be >0. According to the [spec](https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/data_structures.md#state-on-provider-chain), 
> vscId: uint64 is a monotonic strictly increasing and positive ID that is used to uniquely identify the VSCs sent to the consumer chains. Note that 0 is used as a special ID for the mapping from consumer heights to provider heights.

Reply:
Solved by setting a default value for `valset_update_id` in default genesis
https://github.com/cosmos/interchain-security/blob/df266225cc2d88fa704e9e84b7d0505cc65cff52/x/ccv/provider/types/keys.go#L28

https://github.com/cosmos/interchain-security/blob/07377c938016a36b8936ba1825e325aa0d05f03e/x/ccv/provider/types/genesis.go#L34-L39